tests/e2e/test_bridge1_conformance.py

```
@@ -0,0 +1,370 @@
+# ---------------------------------------------------------------------------
+
+def _topology_with_utterance():
+    """Single-satellite topology with recognizer_loop:utterance whitelisted."""
+    b = TopologyBuilder()
+    m = b.add_master("M0")
+    b.add_satellite("S0", upstream=m,
+                    allowed_types=["recognizer_loop:utterance", "speak"])
+    return b
+
+
+def _make_utterance(seq: int = 0, session_id: str = None, lang: str = "en-US") -> Message:
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)